modules/standard/Random.chpl

```diff
@@ -352,17 +375,15 @@ module Random {
   {
     import Search;
     import Sort;
-    
+
     if prob.size != X.size {
       throw new owned IllegalArgumentError('choice() x.size must be equal to prob.size');
     }
-    
+
     if prob.size == 0 then
       throw new owned IllegalArgumentError('choice() prob array cannot be empty');
 
-    const low = X.alignedLow,
-          stride = abs(X.stride);
-    ref P = prob.reindex(0..<X.size);
+    ref P = reshape(prob, {0..<X.size});
```

Comment:
I don't have a problem with this, but, is there any reason you are not keeping the previous idiom? (`prob.reindex(bla)`)